### PR TITLE
Feature/icon component refactor mike

### PR DESCRIPTION
**Pull request by MikeJalow**

### CODE REVIEW

File: component-library/component-lib/src/lib/shared/icon/icon.component.ts [hunk 1]
```diff
@@ -20,21 +11,36 @@ export interface IIconConfig {
...
+                  if(!item.currentValue.ariaLabel){
+                    cont.innerHTML = `
+                    <span class='${item.currentValue.fontFamily}' aria-hidden='true'></span>
+                    `
+                    console.log('changed');
+                  } else {
+                    cont.innerHTML = `
+                    <span class='${item.currentValue.fontFamily}' aria-label='${item.currentValue.ariaLabel}'></span>
+                    `
+                    console.log('also changed');
```
Comment (kristopher-houston) [inline component-library/component-lib/src/lib/shared/icon/icon.component.ts:39]: Remove console logs

**Review by kristopher-houston — COMMENTED** [review]
I'd like to go over this one

**Comment by xingruoxicic** [comment]
https://user-images.githubusercontent.com/111757720/230105958-d4b31289-4578-439f-808c-5b67bed155e1.mov

Working great!

**Review by xingruoxicic — APPROVED** [review]

**Comment by xingruoxicic** [comment]
@MikeJalow You can cherry pick `bde8b98fabcfba53d14cc29688ea3022d4ff15f8` to fix the syntax issues next.

File: component-library/component-lib/src/lib/shared/icon/icon.component.ts [hunk 2]
```diff
@@ -1,17 +1,8 @@
...
-    ['fa-regular']: null,
-    ['fa-brands']: null,
-};
+import { Component, Input, KeyValueDiffer, KeyValueDiffers } from '@angular/core';
 
 export interface IIconConfig {
     ariaLabel?: string;
-    unicode: string;
-    fontFamily: keyof typeof FONT_FAMILIES;
+    fontFamily?: string;
```
Comment (kristopher-houston) [inline component-library/component-lib/src/lib/shared/icon/icon.component.ts:5]: Why was the type of this changed?

**Review by kristopher-houston — APPROVED** [review]

**Review by xingruoxicic — APPROVED** [review]
Please remove all console logs and we can call it a day

**Pull request merged**